### PR TITLE
Nz/3405 Allow user to edit HITECH APD name

### LINKED ISSUE #3405: [Feature] Allow user to edit HITECH APD name
Design task #3308

**Description and related issues -**
Currently our APDs all have the same name. We need to add a field to the APD Overview to all for the user to name an APD

1) Add APD Name field on the ADP Overview page 
2) Change the header design of the APD Name and the designation of HITECH IAPD | FFY XXXX - XXXX as shown in Figma
3) When the user types in the APD Name, the top of the page will populate, just like it does when the name of an Activity is added or edit on the Activities page. 

4) Allow for editing the APD Name at the top of each page where the APD Name is currently shown. 

**Mockups or link to Figma**
https://www.figma.com/file/6eVvo7JjvXiovGTR4BoVgK/CMS-eAPD-2021?node-id=4509%3A30771

### This task is done when…

- [x] updated create APD endpoint so that the default name is Untitled APD
- [x] updated the APD patch endpoint to allow updating the name field
- [x] added APD Name field on ADP Overview page
- [x] updated the APD header to have an edit button (see Figma) that changes the APD Name to an editable APD Name field
- [x] changing the name in either place updates it everywhere

### PR DESCRIPTION
**Pull request by Sun-Mountain**
Resolves #3405 

**Description-**

Currently our APDs all have the same name. We need to add a field to the APD Overview to all for the user to name an APD.

**This pull request changes...**

- Allows state staff user to edit APD name through text box in APD summary, inline editor in the APD header. (Users can also press an `edit` button in the APD header.)
- Test for changing APD name
- Add `EDIT_APD_NAME` symbol to update state when name changes.

**This pull request also touches…**

- refactored styling so that if a dev needs to remove `clearfix` (which is default in the cms style guide), they only need to add `className="remove-clearfix"`.
- Remove `/name` from `staticFields`
- Modified APD header div ids

**Steps to manually verify this change...**

1. Sign in as state staff
2. In an APD, change the name of the APD thru
    a. APD summary textbox
    b. APD header inline text edit
    c. clicking the edit button in the APD header
3. Confirm APD name has changed in each iteration

### This pull request is ready to review when...

- [x] Automated tests are updated (and all tests are passing)
- [ ] The change has been documented
- [ ] Associated OpenAPI documentation has been updated
- [ ] Changelog is updated as appropriate
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This pull request can be merged when…

- [x] Code has been reviewed by someone other than the original author
- [x] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: web/src/actions/editApd/overview.js [hunk 1]
```diff
@@ -1,6 +1,19 @@
...
 
+/**
+ * Rename an activity
+ * @param {Number} index The index of the activity to rename
+ * @param {String} name The new activity name
+ */
+ export const setActivityName = (index, name) => (dispatch, getState) => {
+  dispatch({
+    type: EDIT_APD,
+    path: `/apd/${index}/name`,
```
Comment (thetif) [inline web/src/actions/editApd/overview.js:12]: I see what's happening. In this case, path is a json path. We are using jsonpatch http://jsonpatch.com/. This dispatch is setting out an EDIT_APD message which is picked up by the apd reducer, which updates that path in the json with the value you sent in.

**Review by tbolt — APPROVED** [review]
Looks good. Works as specified. 

One note- I do wonder about the accessibility of the editable field. When I used VoiceOver it recognized both the field and the "Edit" as buttons. It should probably be an <input> if possible. Also there wasn't any label to it so VoiceOver just said "button" rather than "APD Name." Maybe we could add a label with `class="ds-u-visibility--screen-reader"`?

File: web/src/actions/editApd/overview.js [hunk 2]
```diff
@@ -1,6 +1,24 @@
-import { ADD_APD_YEAR, EDIT_APD, REMOVE_APD_YEAR } from './symbols';
+import { ADD_APD_YEAR, EDIT_APD, EDIT_APD_NAME, REMOVE_APD_YEAR } from './symbols';
 import { updateBudget } from '../budget';
 
+/**
+ * Rename an activity
+ * @param {String} name The new activity name
+ */
+ export const setApdName = (name) => (dispatch, getState) => {
```
[the remark this answers is not in the document]
Comment (Sun-Mountain) [reply web/src/actions/editApd/overview.js:8]: @thetif might be able to explain it better, but one updates the name and the other retrieves the updated name and updates State.

[the remark this answers is not in the document]
Comment (thetif) [reply web/src/actions/editApd/overview.js:8]: Yes, EDIT_APD saves it in the backend and EDIT_APD_NAME updates it in the APD list on the dashboard.

**Comment by thetif** [comment]
CONFIRMED:

|Given|When|Then|Result|
|---|---|---|---|
|editing using the APD Name field in APD Overview|changing the name|the name also changes in the header|:white_check_mark: PASS|
|editing using the APD Name field in APD Overview|changing the name|saving to the backend is triggered|:white_check_mark: PASS|
|editing using the APD Name field in APD Overview|changing the name|the name is also changed in the dashboard view|:white_check_mark: PASS|
|the APD name in the header|clicking the pencil icon|the name turns into an input field that allows editing|:white_check_mark: PASS|
|the APD name in the header|clicking the name|the name turns into an input field that allows editing|:white_check_mark: PASS|
|editing using the APD name field in the header|changing the name|the name also change in the header|:white_check_mark: PASS|
|editing using the APD name field in the header|changing the name|saving to the backend is triggered|:white_check_mark: PASS|
|editing using the APD name field in the header|changing the name|the name is also changed in the dashboard view|:white_check_mark: PASS|

**Review by thetif — COMMENTED** [review]
everything looks good

**Review by thetif — APPROVED** [review]

**Review by aricka-lewis — APPROVED** [review]
Everything looks good to me

**Review by NAretakis — APPROVED** [review]
Yes! Works great, approved. Nice work.

**Pull request merged**